annotations/annotations_rw.go

```diff
@@ -13,6 +14,7 @@ import (
 
 type RW interface {
 	Read(ctx context.Context, contentUUID string) ([]*Annotation, bool, error)
+	Write(ctx context.Context, contentUUID string, annotations []Annotation) error
```

Comment:
❓ The asymmetry that the Read returns `[]*Annotation` but the Write accepts `[]Annotation` seems a bit odd.

Reply:
The reason why I used `[]Annotation` in `Write` it is because the `Canonicalizer` returns such data type.
I am happy to make it symmetric. The way that I can make it symmetric depends how much we care of CPU and memory efficiency.
If we care, `Write` should receive a `[]*Annotation` and I need to modify the `Canonicalizer`.
If we do not care so much of efficiency, `Read` should return `[]Annotation` and I need to modify the annotations augmenter.

Do you have preferences?

Reply:
My preference is for `[]Annotation` just because I think it's simpler and these are straightforward value objects.

Reply:
I made the interface consistent by returning a `[]Annotation` in the `Read` method